ethereumj-core/src/main/java/org/ethereum/crypto/HashUtil.java

```diff
@@ -1,35 +1,58 @@
 package org.ethereum.crypto;
 
-import org.ethereum.crypto.cryptohash.Keccak256;
-import org.ethereum.crypto.cryptohash.Keccak512;
+import org.ethereum.config.SystemProperties;
+import org.ethereum.crypto.jce.SpongyCastleProvider;
 import org.ethereum.util.RLP;
 import org.ethereum.util.Utils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 import org.spongycastle.crypto.Digest;
 import org.spongycastle.crypto.digests.RIPEMD160Digest;
 import org.spongycastle.util.encoders.Hex;
-
+import org.springframework.util.StringUtils;
 import java.math.BigInteger;
 import java.security.MessageDigest;
 import java.security.NoSuchAlgorithmException;
+import java.security.Provider;
+import java.security.Security;
 import java.util.Random;
 
 import static java.util.Arrays.copyOfRange;
 import static org.ethereum.util.ByteUtil.EMPTY_BYTE_ARRAY;
 
 public class HashUtil {
 
-    public static final byte[] EMPTY_DATA_HASH = sha3(EMPTY_BYTE_ARRAY);
-    public static final byte[] EMPTY_LIST_HASH = sha3(RLP.encodeList());
-    public static final byte[] EMPTY_TRIE_HASH = sha3(RLP.encodeElement(EMPTY_BYTE_ARRAY));
-
+	private static final Logger LOG = LoggerFactory.getLogger(HashUtil.class);
+	
+    public static final byte[] EMPTY_DATA_HASH;
+    public static final byte[] EMPTY_LIST_HASH;
+    public static final byte[] EMPTY_TRIE_HASH;
+
+    private static final Provider CRYPTO_PROVIDER;
+    private static final String HASH_256_ALGORITHM_DEFAULT_NAME = "ETH-KECCAK-256";
+    private static final String HASH_512_ALGORITHM_DEFAULT_NAME = "ETH-KECCAK-512";
+        
+    private static final String HASH_256_ALGORITHM_NAME;
+    private static final String HASH_512_ALGORITHM_NAME;
+    
     private static final MessageDigest sha256digest;
 
     static {
+    	SystemProperties props = SystemProperties.getDefault();
```

Comment:
We may leave static `SystemProperties` access here for now, but normally we retrieving this bean from the Spring AppContext